apps/rush-lib/src/logic/deploy/DeployManager.ts

```diff
@@ -133,85 +136,131 @@ export class DeployManager {
-        path.join(packageJsonRealFolderPath, 'package.json')
-      );
+    deployState.foldersToCopy.add(packageJsonRealFolderPath);
 
-      // Transform packageJson using pnpmfile.js
-      const packageJson: IPackageJson = deployState.pnpmfileConfiguration.transform(originalPackageJson);
+    const originalPackageJson: IPackageJson = JsonFile.load(
+      path.join(packageJsonRealFolderPath, 'package.json')
+    );
 
-      // Union of keys from regular dependencies, peerDependencies, optionalDependencies
-      // (and possibly devDependencies if includeDevDependencies=true)
-      const allDependencyNames: Set<string> = new Set<string>();
+    const sourceFolderInfo: IFolderInfo | undefined = deployState.folderInfosByPath.get(
+      FileSystem.getRealPath(packageJsonFolderPath)
+    );
+
+    // Transform packageJson using pnpmfile.js
+    const packageJson: IPackageJson = deployState.pnpmfileConfiguration.transform(originalPackageJson);
 
-      // Just the keys from optionalDependencies and peerDependencies
-      const optionalDependencyNames: Set<string> = new Set<string>();
+    // Union of keys from regular dependencies, peerDependencies, optionalDependencies
+    // (and possibly devDependencies if includeDevDependencies=true)
+    const allDependencyNames: Set<string> = new Set<string>();
 
-      for (const name of Object.keys(packageJson.dependencies || {})) {
+    // Just the keys from optionalDependencies and peerDependencies
+    const optionalDependencyNames: Set<string> = new Set<string>();
+
+    for (const name of Object.keys(packageJson.dependencies || {})) {
+      allDependencyNames.add(name);
+    }
+    if (deployState.scenarioConfiguration.json.includeDevDependencies) {
+      for (const name of Object.keys(packageJson.devDependencies || {})) {
         allDependencyNames.add(name);
       }
-      if (deployState.scenarioConfiguration.json.includeDevDependencies) {
-        for (const name of Object.keys(packageJson.devDependencies || {})) {
-          allDependencyNames.add(name);
+    }
+    for (const name of Object.keys(packageJson.peerDependencies || {})) {
+      allDependencyNames.add(name);
+      optionalDependencyNames.add(name); // consider peers optional, since they are so frequently broken
+    }
+    for (const name of Object.keys(packageJson.optionalDependencies || {})) {
+      allDependencyNames.add(name);
+      optionalDependencyNames.add(name);
+    }
+
+    if (sourceFolderInfo && sourceFolderInfo.isRushProject) {
+      const projectSettings: IDeployScenarioProjectJson | undefined =
+        sourceFolderInfo && sourceFolderInfo.projectSettings;
+      const additionalDependenciesToInclude: string[] | undefined =
+        projectSettings && projectSettings.additionalDependenciesToInclude;
+      const dependenciesToExclude: string[] | undefined =
+        projectSettings && projectSettings.dependenciesToExclude;
+
+      this._applyDependencyFilters(
+        allDependencyNames,
```

Comment:
I end up mutating `allDependencyNames`, I hope that is okay

Reply:
I renamed it to `dependencyNamesToProcess` to clarify that it's no longer "all" dependencies

Reply:
I like that change. I considered making a new variable as well but couldn't decide on a name I liked 😂